### PR TITLE
Improve build

### PR DESCRIPTION
**Pull request by tellowkrinkle**
- Use `./autogen.sh` to generate SDL2_image's configure file in build
  - Previously the build would fail because this was missing
- Compile SMPEG with `-std=C++03` because it errors in C++11
- Properly compile the final executable with the right macOS deployment target
  - Previously all the libs were compiled with `-mmacosx-version-min=10.5` but the final binary wasn't, making it all pointless
- Increase the macOS deployment target to 10.7
  - Xcode 10 and up no longer allow compiling for macOS versions that need libstdc++
  - If you're using Xcode 9 or below, you can compile for 10.5 with `make MACOSX_DEPLOYMENT_TARGET=10.5`
- Add enough dependencies to the makefile that you can now compile with `-j8` without it breaking
- Fix the issue where internal lib builds would pickup homebrew SDL
  - Was due to them finding it with pkg-config.  Normally you'd fix this by setting a PKG_CONFIG_PATH but the SDL pkg-config file is also missing the static library list so it's better to pretend pkg-config doesn't exist at all

### CODE REVIEW

File: src/extlib/src/smpeg-2.0.0/configure.in [hunk 1]
```diff
@@ -61,6 +61,11 @@ AC_PROG_INSTALL
 dnl Work around the lack of certain typedefs.
 AC_TYPE_SOCKLEN_T
 
+dnl The script previously overwrote CXXFLAGS with CFLAGS, but that's not always okay
+dnl Save CFLAGS here, then clear it, and finally use it to append to both
+save_CFLAGS="$CFLAGS"
+CFLAGS=""
+
```
Comment (DanielOaks) [inline src/extlib/src/smpeg-2.0.0/configure.in]: Just curious, is this also upstream or when we update smpeg will we need to re-add these changes on top again?

File: src/extlib/src/smpeg-2.0.0/configure.in [hunk 2]
```diff
@@ -210,11 +215,11 @@ CFLAGS="$CFLAGS -Wall"
...
 CFLAGS="$CFLAGS -I.. -DNOCONTROLS"
 CFLAGS="$CFLAGS -I\$(top_srcdir) -I\$(top_srcdir)/audio -I\$(top_srcdir)/video"
-CXXFLAGS="$CFLAGS"
 
 
-dnl C++ flags are the same as the C flags
-CXXFLAGS="$CFLAGS"
+CXXFLAGS="$CXXFLAGS -std=c++03 $CFLAGS"
+dnl Restore pre-save CFLAGS (see above comment on creation of save_CFLAGS)
+CFLAGS="$save_CFLAGS $CFLAGS"
```
Comment (DanielOaks) [inline src/extlib/src/smpeg-2.0.0/configure.in]: same q as above

**Comment by DanielOaks** [comment]
Nice, this is super cool! Honestly I've sorta lost the ability to dig in and review this properly, but given the age of this code and all I'd be pretty happy to merge this in once we answer that smpeg question. Thanks very much for the work mate :D

**Comment by tellowkrinkle** [comment]
Ahh no that's not upstream I should probably look into trying to upstream it

**Comment by tellowkrinkle** [comment]
[PR'd to upstream](http://icculus.org/pipermail/smpeg/2020-March/000192.html) (or whatever the email equivalent of a PR is)
So as long as that gets merged you shouldn't have to worry if you update smpeg

**Comment by DanielOaks** [comment]
Awesome, thanks very much mate! We'll see what they do but either way we'll have this PR to refer to. I'm happy to merge this whenever you think it's ready to go :D

**Comment by tellowkrinkle** [comment]
Yeah that's the first message in 6 years so I'm not sure if anyone's still there but we'll see (if no one's there, there won't be any smpeg updates to worry about so I guess it wouldn't matter)

And it looks like I broke the static lib linux build with my disabling of pkg-config, let me find another solution to this...

**Comment by tellowkrinkle** [comment]
And that one broke my linux steam build
Need to sleep, will try again tomorrow

**Comment by tellowkrinkle** [comment]
Okay seems to be working on Linux with and without steam, macOS, and Windows msys2
So I think I'll call it good

**Comment by euank** [comment]
These changes LGTM, thanks @tellowkrinkle!

I'll leave it to @DanielOaks to merge since they took a look already as well.

**Comment by DanielOaks** [comment]
Awesome, thanks again for the build improvements mate! :D

**Pull request merged**